packages/material-ui/src/BottomNavigation/BottomNavigation.d.ts

```diff
@@ -7,15 +7,29 @@ export interface BottomNavigationProps
     BottomNavigationClassKey,
     'onChange'
   > {
-  children: React.ReactNode;
+  /**
+   * The component used for the root node.
+   * Either a string to use a DOM element or a component.
+   */
   component?: React.ElementType<React.HTMLAttributes<HTMLDivElement>>;
-  onChange?: (event: React.ChangeEvent<{}>, value: any) => void;
+  /**
+   * Callback fired when the value changes.
+   *
+   * @param {object} event The event source of the callback
+   * @param {any} value We default to the index of the child
+   */
+  onChange?(event: React.ChangeEvent<{}>, value: any): void;
```

Comment:
I think that's not quite correct. I would expect that `this` would by of `BottomNavigationProps` type if I specify `onChange`. But it really is an arrow function and the `this` points to the outer one.

Reply:
I changed it so that `event` and `value` would have documentation, I can revert it but they wont have any in-editor documentation if I do.

Reply:
That's weird. I need to check again but I would suspect `{ foo(): void; }` and `{ foo: () => void }` to have different semantics with regard to `this`.

If they're not picked up by vscode then that's probably something we should look into upstream.

Reply:
Yeah I'm not sure about this one. It might work from an assignability point in most cases but it's just not how we define these handlers usually. At runtime they might be bound to any other object.

Reply:
> would expect that `this` would by of BottomNavigationProps type if I specify onChange

Care to elaborate on why that is? We don't set the type of `this` (see below) so I don't see why you'd expect that.

```typescript
onChange?(this: BottomNavigationProps, event: React.ChangeEvent<{}>, value: any): void;
```

Reply:
> Care to elaborate on why that is? 

Sure. Maybe [this code](https://www.typescriptlang.org/play/#code/JYOwLgpgTgZghgYwgAgEIHsCuIAmyDeAUMicjOugBQCUAXMiJgLYBG0A3MaS3FPY6w6EAvoUKhIsRCgCqIFllwEuJcuno1kAXgB8DZmyidSyHn32CjIsQnQgAzmFOKc9DNjxblJtZqImTKAgwTCgQZDAAC2B7ADozY1JhABoVU156AEZrIA) helps. 

Essentially in ` { foo(): void }` inside `foo` `this` is of type `{ foo(): void }` whereas in ` { foo: () => void }` inside `foo` `this` has the type of the outer `this`.

Reply:
I see, though no matter what the type definition says you can implement it both ways.
I wouldn't think of using `this` in a callback that doesn't specify it's type, anyways it's up to you.

Reply:
Let's use the new syntax to improve the docs for almost everyone and then deal with (likely no) breakage later.